src/Adapter/Product/Combination/QueryHandler/SearchCombinationsForAssociationHandler.php

```diff
@@ -101,12 +121,28 @@ private function createResult(array $foundCombination): CombinationForAssociatio
             $imagePath = $this->productImagePathFactory->getNoImagePath(ProductImagePathFactory::IMAGE_TYPE_HOME_DEFAULT);
         }
 
+        $combinationId = (int) ($foundCombination['id_product_attribute'] ?? NoCombinationId::NO_COMBINATION_ID);
+
         return new CombinationForAssociation(
             (int) $foundCombination['id_product'],
-            (int) ($foundCombination['id_product_attribute'] ?? NoCombinationId::NO_COMBINATION_ID),
-            $foundCombination['name'],
+            $combinationId,
+            $this->buildName($foundCombination['name'], $combinationId, $languageId),
             $foundCombination['combination_reference'] ?? ($foundCombination['product_reference'] ?? ''),
             $imagePath
         );
     }
+
+    private function buildName(string $productName, int $combinationId, LanguageId $languageId): string
+    {
+        $builtProductName = $productName;
+        if ($combinationId !== NoCombinationId::NO_COMBINATION_ID) {
+            $attributesInformation = $this->attributeRepository->getAttributesInfoByCombinationIds(
+                [new CombinationId($combinationId)],
+                $languageId
+            );
+            $builtProductName = $this->combinationNameBuilder->buildFullName($builtProductName, $attributesInformation[$combinationId]);
+        }
+
+        return $builtProductName;
+    }
```

Comment:
Not sure you need this copy variable, you can simply modify the existing `$productName` one Or even better use early return:

Reply:
yes early return is always more understandable